src/modules/WKDLocate.js

```diff
@@ -0,0 +1,265 @@
+        }
+      });
+      if (keptUsers.length) {
+        key.users = keptUsers;
+        if (onlyOne) {
+          if (!candidate) {
+            candidate = key;
+          } else { // More then one key in WKD is a rare case. Example can be found as "test-multikey@testkolab.intevation.de"
+            const newValid = key.verifyPrimaryKey() === openpgp.enums.keyStatus.valid;
+            const oldValid = candidate.verifyPrimaryKey() === openpgp.enums.keyStatus.valid;
+            // Prefer the one that is valid
+            if (newValid && !oldValid) {
+              candidate = key;
+              console.log(`WKD: Preferring ${key.primaryKey.getFingerprint()} over ${candidate.primaryKey.getFingerprint()} because of validity.`);
+            } else if (newValid === oldValid && key.primaryKey.created > candidate.primaryKey.created) {
+              // If both are valid or invalid check the creation date
+              console.log(`WKD: Preferring ${key.primaryKey.getFingerprint()} over ${candidate.primaryKey.getFingerprint()} because of cr date of primary.`);
+              candidate = key;
+            }
+          }
+        } else {
+          keptArmoredKeys.push(key.armor());
+        }
+      } else {
+        // Example for this can be found as "test-not-matching@testkolab.intevation.de"
+        console.log(`WKD: skipping not matching key '${key.primaryKey.getFingerprint()}' (bad server)`);
+      }
+    });
+    if (onlyOne && candidate) {
+      return candidate.armor();
+    }
+    if (keptArmoredKeys.length) {
+      return keptArmoredKeys;
+    }
+    throw new Error("WKD: Failed to parse any matching key from the result (bad server)");
+  } catch (e) {
+    throw new Error(`WKD: Error handling keys: '${e}'`);
+  }
+}
+
+/** Adds a size limit on a Response which throws
+ * an error if the limit is surpassed.
+ *
+ * Based on: https://fetch.spec.whatwg.org/
+ *
+ * @param {Response}     response  The fetch response.
+ * @param {Number}       limit     The maximum bytes to read.
+ *
+ * @returns {Uint8array} Array containing the data read.
+ */
+function sizeLimitResponse(response, limit) {
+  if (response.status != 200) {
+    throw new Error(`WKD: Invalid WKD Response ${response.status}:${response.statusText}`);
+  }
+
+  const reader = response.body.getReader();
+  let total = 0;
+  const results = new Array;
+  return pump();
+  function pump() {
+    return reader.read().then(({done, value}) => {
+      if (done) {
+        return openpgp.util.concatUint8Array(results);
+      }
+      total += value.byteLength;
+      results.push(new Uint8Array(value));
+      if (total > limit) {
+        // Example for this can be found as "test-large@testkolab.intevation.de"
+        throw new Error("WKD: Response longer then the max size");
+      }
+      return pump();
+    });
+  }
+}
+
+export default class WKDLocate {
+  constructor() {
+    this.blacklist = prefs.keyserver.wkd_blacklist || [];
```

Comment:
Could do here a `this.blacklist = (prefs.keyserver.wkd_blacklist || []).map(item => RegExp(item, "i"))`
Then this is done only once and not on each lookup.